src/Symfony/Component/Clock/MockClock.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class MockClock implements TimeInterface
```

Comment:
Instead of calling the class by what it should be used for it might make more sense to call the class by what it actually does. In this case that would be a `FrozenClock` which freezes the point in time to a specific one. 

Or to multiple specific ones when the constructor also accepts an array of DateTimeImmutables.

A possible further Implementation might then be a `FreezableClock` that can be frozen multiple times using @theofidry's idea via a `freeze` method.

Reply:
> Instead of calling the class by what it should be used for it might make more sense to call the class by what it actually does. In this case that would be a FrozenClock which freezes the point in time to a specific one.

I prefer keeping MockClock. I think the name says both what it might be use for *and* what it does, thus providing better discoverability (and is consistent with "MockHttpClient".)

> Or to multiple specific ones when the constructor also accepts an array of DateTimeImmutables.

`now()` would then return dates from that sequence? Interesting idea :)

> A possible further Implementation might then be a FreezableClock that can be frozen multiple times using @theofidry's idea via a freeze method.

Agreed. I might wait for someone that needs that to submit a PR instead of providing it in this initial PR though.

Reply:
Hey @nicolas-grekas 

I saw that your first version `MockClock` is not `final`, But I can see that all class are final, what made you add classes as final?

I ask this because I see others classes from symfony implementing an interface But those aren't final.
for example: https://github.com/symfony/serializer/blob/7.0/Encoder/JsonEncode.php#L21

 is there any reason to make those clock classes as final and others not?

Reply:
It's subtle and certainly never a black and white decision. Why are you asking?

Reply:
Because I see some open source projects closing their classes for extensions and force the dev implement the interface, or maybe decarorate the classes in case they want to reuse some existing code.  And as far as I know it is to help maintainers handle BC easily, because the dev won't extend classes. 


But why symfony doesn't close the api and make the dev always implement the interface?

And as this clock classes are different of others,  I thought it has a different reason for this. 

BTW, thank you for your time in replying me, I really appreciate that.

Reply:
In Symfony we're not dogmatic on the topic. I think we learned that from a maintenance perspective, making classes final makes it easier to evolve them without too much BC concerns. But our core desire is to empower end users, and "final" comes into the way sometimes so we're open to reconsidering. As for why some classes are not final: history for some, and openness to inheritance for others.
The open/closed principle is a balanced one.

Reply:
Yes, I mean more flexibility to achieve what they need to achieve, and some framework to guide them in the process :)

Reply:
This vision is so nice,  because even you know that closes classes are easier for maintaining, but symfony prefer give flexibility for users,  even if it costs more work from symfony side 🤯♥️

Well thank you again for your time